app/models/person.rb

```diff
@@ -48,11 +49,16 @@ class Person < ApplicationRecord
 
   # validations
   validates :name, :intellectual_property, presence: true
+  validates :wikidata_uri, format: WIKIDATA_URI_PATTERN, allow_nil: true
 
   validates_attachment_content_type :profile_image, content_type: /\Aimage\/.*\z/
 
   update_index('people'){self} # update PeopleIndex when entity is updated
 
+  before_validation do
+    self.wikidata_uri = wikidata_uri.blank? ? nil : wikidata_uri.strip.downcase
```

Comment:
the Q in https://wikidata.org/wiki/Q1234 is uppercase, canonically. It's true there is an auto-redirect in place at wikidata.org, but it's still incorrect for us to store a lowercase q. Please fix this.

Reply:
Oh, thanks, did not know that.
The problem why specs failing is because I've added wikidata_uri.downcase call in Person.before_validation block.
Just wanted to make regexp validation simpler with 'normalizing' input.

But as URIs are case-sensitive I've added more complex normalization logic to make Q uoppercase while rest of URI is downcased.